src/Symfony/Bundle/FrameworkBundle/Tests/Command/CacheClearCommand/CacheClearCommandTest.php

```diff
@@ -41,65 +41,96 @@ protected function tearDown(): void
+                    $this->fail(sprintf('Meta file "%s" is not fresh', (string) $file));
+                }
+            );
         }
 
         // check that app kernel file present in meta file of container's cache
-        $containerClass = $kernel->getContainer()->getParameter('kernel.container_class');
+        $containerClass = $this->kernel->getContainer()->getParameter('kernel.container_class');
         $containerRef = new \ReflectionClass($containerClass);
         $containerFile = \dirname($containerRef->getFileName(), 2).'/'.$containerClass.'.php';
         $containerMetaFile = $containerFile.'.meta';
-        $kernelRef = new \ReflectionObject($kernel);
-        $kernelFile = $kernelRef->getFileName();
+        $this->kernelRef = new \ReflectionObject($this->kernel);
+        $this->kernelFile = $this->kernelRef->getFileName();
         /** @var ResourceInterface[] $meta */
         $meta = unserialize(file_get_contents($containerMetaFile));
         $found = false;
         foreach ($meta as $resource) {
-            if ((string) $resource === $kernelFile) {
+            if ((string) $resource === $this->kernelFile) {
                 $found = true;
                 break;
             }
         }
         $this->assertTrue($found, 'Kernel file should present as resource');
 
         $containerRef = new \ReflectionClass(require $containerFile);
-        $containerFile = str_replace('tes_'.\DIRECTORY_SEPARATOR, 'test'.\DIRECTORY_SEPARATOR, $containerRef->getFileName());
-        $this->assertMatchesRegularExpression(sprintf('/\'kernel.container_class\'\s*=>\s*\'%s\'/', $containerClass), file_get_contents($containerFile), 'kernel.container_class is properly set on the dumped container');
+        $containerFile = str_replace(
+            'tes_'.\DIRECTORY_SEPARATOR,
+            'test'.\DIRECTORY_SEPARATOR,
+            $containerRef->getFileName()
+        );
+        $this->assertMatchesRegularExpression(
+            sprintf('/\'kernel.container_class\'\s*=>\s*\'%s\'/', $containerClass),
+            file_get_contents($containerFile),
+            'kernel.container_class is properly set on the dumped container'
+        );
     }
 
-    public function getKernel()
+    public function testCacheIsWarmedWhenCalledTwice()
     {
-        yield [new TestAppKernel('test', true)];
-        yield [new NoBuildDirKernel('test', true)];
+        $input = new ArrayInput(['cache:clear']);
+        $application = new Application(clone $this->kernel);
+        $application->setCatchExceptions(false);
+        $application->doRun($input, new NullOutput());
+
+        $_SERVER['REQUEST_TIME'] = time() + 1;
+        $application = new Application(clone $this->kernel);
+        $application->setCatchExceptions(false);
+        $application->doRun($input, new NullOutput());
+
+        $this->assertTrue(is_file($this->kernel->getCacheDir().'/annotations.php'));
     }
-}
 
-class NoBuildDirKernel extends TestAppKernel
```

Comment:
Could you have a look @chalasr ? Previous code can't work, because `kernel.build_dir` is now needed to build the container (and now needed to initialize the `cache:clear` command).
I refactored the code t build a container and then remove the parameter from the dumped container.